modules/gdscript/gdscript.cpp

```diff
@@ -1062,6 +1062,7 @@ void GDScript::set_path(const String &p_path, bool p_take_over) {
 	String old_path = path;
 	path = p_path;
 	path_valid = true;
+	fully_qualified_name = fully_qualified_name.replace(old_path, p_path);
```

Comment:
I think it would be safer to re-evaluate the FQCN rather than use `replace()`. In most cases this should work, but there may be collisions for scripts located outside of `res://` or `class_name`-based FQCNs.